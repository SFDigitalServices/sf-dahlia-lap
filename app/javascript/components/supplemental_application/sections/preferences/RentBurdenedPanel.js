import React from 'react'
import { SelectField, CurrencyField } from '~/utils/form/final_form/Field.js'

import FormGrid from '~/components/molecules/FormGrid'
import { FormItem } from './utils'
import { buildFieldId } from '~/components/applications/application_form/preferences/utils'
import { statusOptions } from '~/components/supplemental_application/sections/preferences/utils'
import formOptions from '~/components/applications/application_form/formOptions'
import validate from '~/utils/form/validations'
const { labelize } = formOptions

export const RentBurdenedPanel = ({ preferenceIndex, visited }) => (
  <FormGrid.Row expand={false}>
    <FormItem label='Preference Name'>
      <div className='text-value'>
        Rent Burdened Housing Preference
      </div>
    </FormItem>
    <FormItem>
      {/* Total monthly rent is stored in the application, not the preference. */}
      <CurrencyField
        fieldName='total_monthly_rent'
        type='number'
        label='Total Household Monthly Rent'
        validation={validate.isValidCurrency('Please enter a valid dollar amount.')}
        isDirty={visited && visited.total_monthly_rent} />
    </FormItem>
    <FormItem>
      <SelectField
        fieldName={buildFieldId(preferenceIndex, 'post_lottery_validation')}
        options={labelize(statusOptions, { disableEmpty: true })}
        label='Status'
        className='preference-status-select' />
    </FormItem>
  </FormGrid.Row>
)

export default RentBurdenedPanel
