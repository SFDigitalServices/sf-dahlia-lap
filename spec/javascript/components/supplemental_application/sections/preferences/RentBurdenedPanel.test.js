/* global shallow */
import React from 'react'

import RentBurdenedPanel from 'components/supplemental_application/sections/preferences/RentBurdenedPanel'

describe('RentBurdenedPanel', () => {
  test('should render successfully', () => {
    const wrapper = shallow(<RentBurdenedPanel preferenceIndex='1' />)

    expect(wrapper).toMatchSnapshot()
  })
})
