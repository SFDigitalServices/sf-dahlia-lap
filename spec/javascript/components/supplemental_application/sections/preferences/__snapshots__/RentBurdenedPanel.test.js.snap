// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RentBurdenedPanel should render successfully 1`] = `
<Component
  expand={false}
>
  <FormItem
    label="Preference Name"
  >
    <div
      className="text-value"
    >
      Rent Burdened Housing Preference
    </div>
  </FormItem>
  <FormItem>
    <CurrencyField
      fieldName="total_monthly_rent"
      label="Total Household Monthly Rent"
      type="number"
      validation={[Function]}
    />
  </FormItem>
  <FormItem>
    <SelectField
      className="preference-status-select"
      fieldName="preferences.1.post_lottery_validation"
      label="Status"
      options={
        Array [
          Object {
            "disabled": "disabled",
            "label": "Select One...",
            "value": "",
          },
          Object {
            "label": "Unconfirmed",
            "value": "Unconfirmed",
          },
          Object {
            "label": "Confirmed",
            "value": "Confirmed",
          },
          Object {
            "label": "Invalid",
            "value": "Invalid",
          },
        ]
      }
    />
  </FormItem>
</Component>
`;
